Fix/fixing build errors - NOT READY FOR MERGE

mateodelnorte opened the pull request:
NOT READY FOR MERGE. 

- fixed ios build errors. 
- removed libTcpSocket references from XCode project build and link settings. 
- react-native upgraded project. 
- re-initialized project and re-merged back in appropriate changes. note android is not currently working. 
- consolidated ios and android index files to call common dep to load app. 

IOS fix `npm run clean && npm run install && npm run ios`es and runs correctly, repeatedly. 

I'll continue with the android fix and bringing tests back in, in morning.

Commit:
making npm run clean a little heavier. tidying up any commands speficying ./node_modules. no need to type the full path for any executable that appears in ./node_modules/.bin. any command specified in the "bin" section of a child dependency will get linked into there and npm will find them automatically

Commit:
Merge branch 'master' of github.com:GridPlus/gridplus-login into feature/tidying-up-npm-scripts

Commit:
react-native upgrade changes

Commit:
fixed ios build errors. removed libTcpSocket references from XCode project build and link settings. upgraded project. re-initialized project and re-merged back in appropriate changes. note android is not currently working. I'll continue with the android fix in morning.

alex-miller-0 commented:
Still running into the randombytes error (`undefined is not an object (evaluating at RNRandomBytes.seed`) :(

alex-miller-0 committed:
Exports sagas modules as variables.
This fixes our issue with rgenerator per this comment: https://github.com/facebook/react-native/issues/14838#issuecomment-313398604
A new error is presented, which is present in thsi commit: nativePerformance.now is undefined.
